src/opm/output/eclipse/AggregateConnectionData.cpp

```diff
@@ -173,8 +173,12 @@ namespace {
 
             sConn[Ix::item12] = sConn[Ix::ConnTrans];
 
-            sConn[Ix::SegDistEnd]   = scprop(M::length, conn.getSegDistEnd());
-            sConn[Ix::SegDistStart] = scprop(M::length, conn.getSegDistStart());
+
+            if (conn.attachedToSegment()) {
+                const auto& perf_range = conn.perf_range();
+                sConn[Ix::SegDistStart] = scprop(M::length, perf_range->first);
+                sConn[Ix::SegDistEnd]   = scprop(M::length, perf_range->second);
```

Comment:
If you don't want to refer to `->first` and `->second` you can use structured bindings here
```C++
const auto& [start, end] = conn.perf_range();
sConn[Ix::SegDistStart]  = scprop(M::length, start);
sConn[Ix::SegDistEnd]    = scprop(M::length, end);
```

Reply:
Cool - that I did not know about.

Reply:
Structured bindings were introduced in C++17 and they definitely simplify working with `std::pair` (and `std::tuple`).  On the other hand, I don't actually know if they see through `std::optional` so you **may** need to stick to `->first` and `->second` anyway.  Sorry for not thinking through the context here.

Reply:
```C++
const auto& [start, end] = *conn.perf_range();
...
```
worked!